src/Events/QR/QR_Code.php

```diff
@@ -200,15 +212,33 @@ private function delete_qr_image( int $post_id, int $size ): void {
 	 */
 	public function render_modal(): void {
 		$post = get_post( tec_get_request_var( 'post_id' ) );
-		if ( ! $post || ! tribe_is_event( $post ) ) {
-			wp_die( esc_html__( 'Invalid event.', 'the-events-calendar' ) );
+		if ( ! $post ) {
+			wp_die( esc_html__( 'No post found.', 'the-events-calendar' ) );
+		}
+
+		if ( has_action( 'tribe_common_loaded', 'tribe_register_pro' ) ) {
```

Comment:
I was going to let this go for now, but since there is a chunk you need to move in XXX you should move this as well. Add an action here and pass `$post` - have ECP hook into the action, test `$post` and die if necessary.

OR

Add a filter here at the end with a boolean that dies if true and have ECP hook in and trigger it as necessary.